gno.land/pkg/keyscli/run.go

```diff
@@ -113,11 +114,33 @@ func execMakeRun(cfg *MakeRunCfg, args []string, cmdio commands.IO) error {
 	// Set to empty; this will be automatically set by the VM keeper.
 	memPkg.Path = ""
 
-	// construct msg & tx and marshal.
 	msg := vm.MsgRun{
 		Caller:  caller,
 		Package: memPkg,
 	}
+
+	// if a sponsor onchain address is specified
+	if cfg.RootCfg.Sponsor != "" {
```

Comment:
We don’t need to know the sponsor address, just that this will be a sponsored tx.